src/auth/auth-api-request.ts

```diff
@@ -1145,6 +1162,34 @@ export abstract class AbstractAuthRequestHandler {
       request.localId = request.uid;
       delete request.uid;
     }
+    // Construct mfa related user data.
+    if (validator.isNonNullObject(request.multiFactor)) {
+      if (validator.isArray(request.multiFactor.enrolledFactors)) {
+        request.mfaInfo = [];
+        try {
+          request.multiFactor.enrolledFactors.forEach((multiFactorInfo: any) => {
+            // Enrollment time and uid are not allowed for signupNewUser endpoint.
+            // They will automatically be provisioned server side.
+            if (multiFactorInfo.enrollmentTime) {
+              throw new FirebaseAuthError(
+                AuthClientErrorCode.INVALID_ARGUMENT,
+                '"enrollmentTime" is not supported when adding second factors via "createUser()"');
+            } else if (multiFactorInfo.uid) {
+              throw new FirebaseAuthError(
+                AuthClientErrorCode.INVALID_ARGUMENT,
+                '"uid" is not supported when adding second factors via "createUser()"');
+            }
+            request.mfaInfo.push(convertMultiFactorInfoToServerFormat(multiFactorInfo));
+          });
+        } catch (e) {
+          return Promise.reject(e);
+        }
+        if (request.mfaInfo.length === 0) {
```

Comment:
Nit: To keep the mutations to `request` to a minimum, you can do this:

```
const mfaInfo = [];
// Populate mfaInfo

if (request.mfaInfo.length > 0) {
  // You can drop the if condition, if you checked for isNonEmptyArray earlier.
  request.mfaInfo = mfaInfo;
}
```

Reply:
Done